h/api/search/query.py

```diff
@@ -163,3 +163,17 @@ def __call__(self, params):
         }
         del params["any"]
         return result
+
+
+class TagsMatcher(object):
+
+    """Matches the tags field against 'tag' or 'tags' parameters."""
+
+    def __call__(self, params):
+        tags = set(params.getall('tag') + params.getall('tags'))
```

Comment:
Sorry to come back to this but I have one minor nitpick. Could you change this line to: `tags = set(v for k, v in params.items() if k in ['tag', 'tags'])`?

That way it's consistent with the rest of this file, which assumes that `params` is a dict-like iterable, not specifically a `MultiDict`.